File: 0997-find-the-town-judge/0997-find-the-town-judge.js
/**
 * @param {number} n
 * @param {number[][]} trust
 * @return {number}
 */
var findJudge = function(n, trust) {
    let indegree=new Array(n).fill(0);
    let outdegree=new Array(n).fill(0);
    for(let it of trust){
        // console.log(it);
        indegree[it[1]-1]++;
        outdegree[it[0]-1]++;
    }
    for(let i=0;i<n;i++){
        // console.log(indegree[i]);
        // console.log(outdegree[i]);
        if(indegree[i]===n-1 && outdegree[i]===0)return i+1;
    }
    return -1;
};

File: 0997-find-the-town-judge/0997-find-the-town-judge.cpp
class Solution {
public:
    int findJudge(int n, vector<vector<int>>& trust) {
        // vector<int> adj[n];
        vector<int> indegree(n),outdegree(n);
        for(auto it:trust){
            // adj[it[0]].push_back(it[1]);
            indegree[it[1]-1]++;
            outdegree[it[0]-1]++;
        }
        for(int i=0;i<n;i++){
            if(indegree[i]==n-1 && outdegree[i]==0){
                return i+1;
            }
        }
        return -1;
    }
};